app/models/laa_reference.rb

```diff
@@ -9,6 +9,18 @@ def unlink!
     update!(linked: false)
   end
 
+  def adjust_link_and_save!
+    # Unlink the previous laa reference
+    laa_ref_already_liked = LaaReference.find_by(maat_reference: maat_reference, linked: true)
+    laa_ref_already_liked.unlink! if laa_ref_already_liked.present?
+
+    self.linked = true
+
+    save!
+  rescue ActiveRecord::ActiveRecordError => e
+    Sentry.capture_exception(e)
+  end
```

Comment:
This is the method that fixes the issue:
The key concept is that the `find_by` no longer looks for a certain defendant_id. Any linked maat_ref must be unlinked, and then the current maat_ref can be linked.
In this way, we prevent the uniqueness validation (https://github.com/ministryofjustice/laa-court-data-adaptor/blob/d6e347f8d10f284bb943bc66fa6743a006de8994/app/models/laa_reference.rb#L5)

Reply:
I was more thinking of the unlink process in the scenario where we are unlinking the maat_id from a different defendant?